cel/env.go

```diff
@@ -175,23 +182,21 @@ type env struct {
 	pkg          packages.Packager
 	provider     ref.TypeProvider
 	adapter      ref.TypeAdapter
+	chk          *checker.Env
```

Comment:
This is not race safe I think. There is a list of scopes inside declarations inside checker environment. The list is mutable by Push/Pop. These mutations prevent parallel use of the checker environment.

For my use case, this is fine, but let's document that. We also need to make sure that Push/Pops are balanced.

Reply:
Ah, good point, I forgot how this interacts with comprehensions. For the most part the push / pop is used only during environment configuration. Let me work on refactoring this a bit more to be thread-safe since the perf change is substantial, and some projects will really want / need to be able to use the same environment concurrently.